tuf/api/metadata.py

```diff
@@ -1258,17 +1259,122 @@ def to_dict(self) -> Dict[str, Any]:
+    to a variable number of bins, the names of which are generated based on the
+    two class attributes.
+
+    Args:
+        hash_prefix_len: Number of bits between 1 and 32 used to separate the
+            different bins.
+        bin_name_prefix: Prefix of all bin names.
+
+    Raises:
+            ValueError, TypeError, AttributeError: Invalid arguments.
+    """
+
+    def __init__(
+        self,
+        hash_prefix_len: int,
+        bin_name_prefix: str,
+        unrecognized_fields: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        if hash_prefix_len <= 0 or hash_prefix_len > 32:
+            raise ValueError("hash_prefix_len must be between 1 and 32")
+        if not isinstance(bin_name_prefix, str):
+            raise ValueError("bin_name_prefix must be a string")
+
+        self.hash_prefix_len = hash_prefix_len
+        self.bin_name_prefix = bin_name_prefix
+        self.unrecognized_fields: Mapping[str, Any] = unrecognized_fields or {}
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, SuccinctHashDelegations):
+            return False
+
+        return (
+            self.hash_prefix_len == other.hash_prefix_len
+            and self.bin_name_prefix == other.bin_name_prefix
+            and self.unrecognized_fields == other.unrecognized_fields
+        )
+
+    @classmethod
+    def from_dict(
+        cls, succinct_hash_info: Dict[str, Any]
+    ) -> "SuccinctHashDelegations":
+        """Creates ``SuccinctHashDelegations`` object from its json/dict
+        representation.
+
+        Raises:
+            ValueError, KeyError, TypeError, AttributeError: Invalid arguments.
+        """
+        hash_prefix_len = succinct_hash_info.pop("delegation_hash_prefix_len")
+        bin_name_prefix = succinct_hash_info.pop("bin_name_prefix")
+        # All fields left in the succinct_hash_info are unrecognized.
+        return cls(hash_prefix_len, bin_name_prefix, succinct_hash_info)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Returns the dict representation of self."""
+        return {
+            "delegation_hash_prefix_len": self.hash_prefix_len,
+            "bin_name_prefix": self.bin_name_prefix,
+            **self.unrecognized_fields,
+        }
+
+    def get_first_bin_name(self) -> str:
+        """Return the name of the first bin"""
+        return f"{self.bin_name_prefix}-0"
+
+    def _calculate_delegation(self, hash_bits_representation: str) -> str:
+        """Helper function for find_delegation calculating the actual rolename.
+
+        Args:
+            hash_bits_representation: binary bit representation of the target
+                hash.
+        """
+
+        bit_length = self.hash_prefix_len
+        # Get the first bit_length of bits and then cast them to decimal.
+        bin_number = int(hash_bits_representation[:bit_length], 2)
+
+        name_prefix = self.bin_name_prefix
+        return f"{name_prefix}-{bin_number}"
```

Comment:
This does not seem to be in line with the TAP examples and if I recall correctly the previous implementation: 

TAP definition:
> The name will be structured as NAME_PREFIX-COUNT where COUNT is a hexadecimal value between 0 and 2^BIT_LENGTH-1 (inclusive) that represents the bin number

TAP example:
>      "name": alice.hbd-00,

 
Your rolename seems to be decimal and not padded with zeros. Is this correct?